python_examples/age_stratified_example/age_stratified_plot.py

```diff
@@ -0,0 +1,176 @@
+#
+# Reads a csv of age stratified data and plots as a bar chart
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+# csv input files should have the column headers:
+# time, infection_Status1, infection_status2, ..., age_range
+# so there will be multiple entries for each timepoint.
+
+
+class Plotter():
+    """Funtion to take a csv file and return various plots,
+    including the capability to make an age-statified
+    bar chart.
+    """
+    def __init__(self, filepath: str, start_date=None, sum_weekly=True,
+                 age_list: list = None):
+        """Initialise the plotter with a filepath and read
+        data from the csv.
+        Parameters
+        ----------
+        filepath : str
+            Filepath to the .csv containing output data
+        start_date : str
+            Starting date for the simulation, "day-month-year"
+        sum_weekly : bool
+            Flag to plot either each timepoint, or the 7 day sum
+        age_list : list
+            List of the explicit age ranges saved in the csv
+        """
+        if not os.path.splitext(filepath)[1] == ".csv":
+            raise TypeError("input file" + filepath + "must be .csv")
+        self.data = pd.read_csv(filepath)
+        self.age_list = age_list
+        self.start_date = start_date
+        self.sum_weekly = sum_weekly
+
+        # Identify the column containing age stratifcation as any
+        # with the substring "age"
+        self.age_name = next((s for s in list(self.data.columns)
+                             if "age" in s.lower()), None)
+        self.do_ages = (self.age_name is not None)
+        if self.do_ages and (self.age_list is None):
+            num = self.data[self.age_name].max()
+            self.age_list = [str(5*i)+"-"+str(5*i+5) for i in range(num+1)]
+
+    def sum_infectious(self) -> None:
+        """Helper function which sums across all columns containing infectious
```

Comment:
Is this helper function intended to be used externally? Consider marking it as private if not